velox/functions/prestosql/aggregates/DecimalAggregate.h

```diff
@@ -48,21 +50,33 @@ struct LongDecimalWithOverflowState {
     common::OutputByteStream outStream(outputBuffer);
     outStream.append((char*)&count, sizeof(int64_t));
     outStream.append((char*)&overflow, sizeof(int64_t));
-    uint64_t lower = LOWER(sum);
-    int64_t upper = UPPER(sum);
-    outStream.append((char*)&lower, sizeof(int64_t));
-    outStream.append((char*)&upper, sizeof(int64_t));
+    outStream.append((char*)&lowerSum, sizeof(int64_t));
+    outStream.append((char*)&upperSum, sizeof(int64_t));
   }
 
   /*
-   * Total size = sizeOf(count) + sizeOf(overflow) + sizeOf(sum)
-   *            = 8 + 8 + 16 = 32.
+   * Total size = sizeOf(count) + sizeOf(overflow) +
+   *              sizeOf(upperSum) + sizeOf(lowerSum)
+   *            = 8 + 8 + 8 + 8 = 32.
    */
   inline static size_t serializedSize() {
     return sizeof(int64_t) * 4;
   }
 
-  int128_t sum{0};
+  inline int128_t getSum() {
+    return buildInt128(upperSum, lowerSum);
+  }
+
+  inline void setSum(int128_t& sum) {
+    upperSum = UPPER(sum);
+    lowerSum = LOWER(sum);
+  }
+
+  // The accumulator's sum is int128_t; however, it is maintained as two int64_t
+  // values for cross-platform compatibility. Having int128_t causes the struct
+  // alignment to be different on certain architectures resulting in a segfault.
```

Comment:
Probably mention that certain operations such as `placement new` and `reinterpret_cast` used in the aggregation framework require int128_t to be aligned. It is not trivial to achieve this alignment given the layout of the aggregates in the RowContainer.

Reply:
I have updated the paragraph to include your root cause analysis, thank you very much!